### DISCUSSION TITLE
Circuit Breaking with KEDA

### DISCUSSION THREAD

**Discussion by kencieszykowski** [kedacore/keda#3917] [Q&A / Need Help] [answered]
This is kind of a strange question; but it does relate to scaling, I promise!

I'm relatively new to using KEDA, but we are trying to accomplish the following:

- If Kafka lag is noticed on Topic A, scale to ceiling (currently happening)
- If Kafka lag crosses a threshold of value / rate on Topic B, scale down to 0 (break the circuit here)
- Once Topic B goes under the threshold, then return to the baseline `minReplicas`

We have Datadog in our environment as well, and I know you can create functions for that Scaler, but it's kind of doing my head in on how this could be accomplished with Keda.  

Any suggestions and or comments would be much appreciated.  Thank you!

**Comment by tomkerkhove** [comment, marked as answer]
Looks like you've opened a feature request in https://github.com/kedacore/keda/issues/3949
